lib/engine/game/g_1880/game.rb

```diff
@@ -323,25 +329,70 @@ def init_minors
           game_minors.map { |minor| G1880::Minor.new(**minor) }
         end
 
+        def setup
+          @fully_funded_corporations = []
+          @full_cap_event = false
+          @foreign_investors_can_build = true
+          setup_building_permits
+          setup_unsaleable_shares
+        end
+
         def p1
           company_by_id('P1')
         end
 
         def status_array(corporation)
-          return [] unless corporation.floated?
+          return unless corporation.floated?
+          return if corporation.minor?
 
           ["Building Permits: #{corporation.building_permits}"]
         end
 
+        def non_floated_corporations
+          @corporations.each { |c| yield c if !c.minor? && !c.floated? }
+        end
+
         def corporation_show_individual_reserved_shares?(_corporation)
           false
         end
 
+        def event_all_shares_available!
+          @log << "-- Event: #{EVENTS_TEXT['all_shares_available'][1]} --"
+          @corporations.each do |c|
+            c.shares.last(5).each { |s| s.buyable = true }
+          end
+        end
+
+        def event_receive_capital!
+          @log << "-- Event: #{EVENTS_TEXT['receive_capital'][1]} --"
+          @corporations.each do |c|
+            next unless c.num_ipo_shares == 5
+
+            amount = c.par_price.price * 5
+            @bank.spend(amount, c)
+            @fully_funded_corporations << c
+            @log << "#{c.name} recieves #{format_currency(amount)}"
+          end
+          @full_cap_event = true
+        end
+
+        def event_fi_stop_operating!
+          @log << "-- Event: #{EVENTS_TEXT['fi_stop_operating'][1]} --"
+          @minors.each(&:unfloat!)
```

Comment:
The minors still need to check for destination after they stop operating.

Reply:
The destination check happens on the corp side. After B3, minors/FIs do not operate, no track lay, no train running. The can only be merged to corps when tracks are connected

Reply:
> As soon as the two railways have at their disposal a joint network (Station Markers are no obstacle
here) the Foreign Investor is joined to the normal company (this happens at the end of the Foreign
Investor's turn).